docs/source/nodes_and_pipelines/nodes.md

```diff
@@ -303,39 +239,97 @@ def split_data(
     Returns:
         Split data.
     """
-    # Loop through data in chunks building up the training and test sets
-    for chunk in data:  # Iterate over the chunks from data
-        full_data = pd.concat(
-            [chunk]
-        )  # Converts the TextFileReader object into list of DataFrames
-        data_train = full_data.sample(
-            frac=parameters["train_fraction"], random_state=parameters["random_state"]
-        )
-        data_test = full_data.drop(data_train.index)
-
-        X_train = data_train.drop(columns=parameters["target_column"])
-        X_test = data_test.drop(columns=parameters["target_column"])
-        y_train = data_train[parameters["target_column"]]
-        y_test = data_test[parameters["target_column"]]
-        yield X_train, X_test, y_train, y_test  # Use yield instead of return to get the generator object
+
+    data_train = data.sample(
+        frac=parameters["train_fraction"], random_state=parameters["random_state"]
+    )
+    data_test = data.drop(data_train.index)
+
+    X_train = data_train.drop(columns=parameters["target_column"])
+    X_test = data_test.drop(columns=parameters["target_column"])
+    y_train = data_train[parameters["target_column"]]
+    y_test = data_test[parameters["target_column"]]
+
+    label_encoder = LabelEncoder()
+    label_encoder.fit(pd.concat([y_train, y_test]))
+    y_train = label_encoder.transform(y_train)
+
+    return X_train, X_test, y_train, y_test
+
+
+def make_predictions(
+    X_train: pd.DataFrame, X_test: pd.DataFrame, y_train: pd.Series
+) -> pd.Series:
+    """Use a DecisionTreeClassifier model to make prediction."""
+    model = DecisionTreeClassifier()
+    model.fit(X_train, y_train)
+
+    for chunk in X_test:
+        y_pred = model.predict(chunk)
+        y_pred = pd.DataFrame(y_pred)
+        yield y_pred
+
+
+def report_accuracy(y_pred: pd.Series, y_test: pd.Series):
+    """Calculates and logs the accuracy.
+
+    Args:
+        y_pred: Predicted target.
+        y_test: True target.
+    """
+    accuracy = accuracy_score(y_test, y_pred)
+    logger = logging.getLogger(__name__)
+    logger.info("Model has accuracy of %.3f on test data.", accuracy)
 ```
 
-We can now `kedro run` in the terminal. The output shows `X_train`, `X_test`, `y_train`, `y_test` saved in chunks:
+
+
+This `ChunkWiseDataset` is a variant of the `pandas.CSVDataset` where the main change is to the `_save` method that append data instead of overwriting. Below is an example of the `ChunkWiseCSVDataset` implementation:
```

Comment:
```suggestion
The `ChunkWiseDataset` is a variant of the `pandas.CSVDataset` where the main change is to the `_save` method that appends data instead of overwriting it. Below is an example of the `ChunkWiseCSVDataset` implementation:
```